validation/views.py

```diff
@@ -484,9 +492,99 @@ def register(request):
+    context = dict(
+        issues=issues,
+        auth=request.user.is_authenticated,
+        is_linguist=user_is_linguist(request.user),
+    )
+    return render(request, "validation/view_issues.html", context)
+
+
+def view_issue_detail(request, issue_id):
+    issue = Issue.objects.filter(id=issue_id).first()
+
+    form = None
+    if issue.recording:
+        form = EditIssueWithRecording(
+            request.POST, initial={"rec_id": str(issue.recording.id)}
+        )
+        rec = Recording.objects.filter(id=issue.recording.id).first()
+        phrase = None
+    if issue.phrase:
+        form = EditIssueWithPhrase(request.POST, initial={"phrase_id": issue.phrase.id})
+        phrase = Phrase.objects.filter(id=issue.phrase.id).first()
+        rec = None
+
+    if request.method == "POST":
+        if form.is_valid():
+            if rec:
+                speaker_code = form.cleaned_data["speaker"]
+                if speaker_code:
+                    speaker = Speaker.objects.filter(code=speaker_code).first()
+                    rec.speaker = speaker
+
+                new_word = form.cleaned_data["phrase"].strip()
+                if new_word:
+                    new_phrase = Phrase.objects.filter(transcription=new_word).first()
+                    if not new_phrase:
+                        new_phrase = Phrase(
+                            field_transcription=new_word,
+                            transcription=new_word,
+                            translation="",
+                            kind="Sentence" if " " in new_word else "Word",
+                            date=datetime.datetime.now(),
+                            modifier=str(request.user),
+                        )
+                        new_phrase.save()
+                    rec.phrase_id = new_phrase.id
+
+                rec.wrong_word = False
+                rec.wrong_speaker = False
+                rec.save()
+
+            elif phrase:
+                transcription = form.cleaned_data["transcription"].strip()
+                translation = form.cleaned_data["translation"].strip()
+
+                if transcription:
+                    phrase.transcription = transcription
+                if translation:
+                    phrase.translation = translation
+
+                phrase.status = "linked"
+                phrase.modifier = str(request.user)
+                phrase.date = datetime.datetime.now()
+                phrase.save()
+
+            issue.status = "Resolved"
+            issue.save()
+            return HttpResponseRedirect("/issues")
+
+    context = dict(
+        issue=issue,
+        form=form,
+        auth=request.user.is_authenticated,
+        is_linguist=user_is_linguist(request.user),
+    )
+    return render(request, "validation/view_issue_detail.html", context)
+
+
+def close_issue(request, issue_id):
+    issue = Issue.objects.filter(id=issue_id).first()
+    issue.status = "Resolved"
```

Comment:
⚠️ there's a minor bug here!  `"Resolved"` is not a valid option but `"resolved"` _is_! It's less error-prone to use the constant instead:

```suggestion
    issue.status = Issue.RESOLVED
```

Reply:
Thank you!!